### PR TITLE
feat: add ruby-mysql2 & rails dev guide

### PR DESCRIPTION
**Pull request by hooopo**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by sykp241095** [comment]
@hooopo Is it necessary to change title to point out they are "two" samples? (mysql2 + ROR)

**Review by sykp241095 — APPROVED** [review]

**Review by sykp241095 — APPROVED** [review]

File: develop/dev-guide-sample-application-ruby-mysql2.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    DATABASE_USER=<user>
+    DATABASE_PASSWORD=<password>
+    DATABASE_NAME=test
+    DATABASE_ENABLE_SSL=true
+    DATABASE_SSL_CA=<downloaded_ssl_ca_path>
+    ```
+
+   > **Note**
+   >
+   > It is recommended to enable TLS connection when using the public endpoint to connect to TiDB Dedicated.
```
Comment (qiancai) [inline develop/dev-guide-sample-application-ruby-mysql2.md:149]:
```suggestion
   > It is recommended to enable TLS connection when using the public endpoint to connect to a TiDB Dedicated cluster.
```

File: develop/dev-guide-sample-application-ruby-mysql2.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+   >
+   > To enable TLS connection, modify `DATABASE_ENABLE_SSL` to `true` and using `DATABASE_SSL_CA` to specify the file path of CA certificate downloaded from the connection dialog.
```
Comment (qiancai) [inline develop/dev-guide-sample-application-ruby-mysql2.md:151]:
```suggestion
   > To enable TLS connection, modify `DATABASE_ENABLE_SSL` to `true` and use `DATABASE_SSL_CA` to specify the file path of CA certificate downloaded from the connection dialog.
```

File: develop/dev-guide-sample-application-ruby-rails.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+# Connect to TiDB with Rails framework and ActiveRecord ORM
+
+TiDB is a MySQL-compatible database, and [Rails](https://github.com/rails/rails) is popular web application framework written in Ruby. This tutorial gives ruby sample code snippets that work with TiDB using Rails framework and ActiveRecord ORM.
+
+In this tutorial, you can learn how to use TiDB and Rails to accomplish the following tasks:
+
+- Set up your environment.
+- Connect to your TiDB cluster using Rails.
+- Build and run your application. Optionally, you can find [sample code snippets](#sample-code-snippets) for basic CRUD operations.
```
Comment (qiancai) [inline develop/dev-guide-sample-application-ruby-rails.md:14]:
```suggestion
- Build and run your application. Optionally, you can find [sample code snippets](#sample-code-snippets) for basic CRUD operations using ActiveRecord ORM.
```

**Comment by qiancai** [comment]
Doc preview:  https://pingcap-docs-website-scaffold-3ty9yhc5h-qiancai.vercel.app/tidb/dev/dev-guide-sample-application-ruby-mysql2

File: develop/dev-guide-sample-application-ruby-rails.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
+---
+title: Connect to TiDB with Rails framework and ActiveRecord ORM
```
[the remark this answers is not in the document]
Comment (hooopo) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: For SEO :) TiDB + Rails 和 TiDB + ActiveRecord 都是用户可能找的组合

File: develop/dev-guide-sample-application-ruby-mysql2.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+For more information, refer to [Delete data](/develop/dev-guide-delete-data.md).
+
+## Best practices
+
+By default, the mysql2 gem can search for existing CA certificates in a particular order until a file is discovered.
+
+1. `/etc/ssl/certs/ca-certificates.crt` # Debian / Ubuntu / Gentoo / Arch / Slackware
+2. `/etc/pki/tls/certs/ca-bundle.crt` # RedHat / Fedora / CentOS / Mageia / Vercel / Netlify
+3. `/etc/ssl/ca-bundle.pem` # OpenSUSE
+4. `/etc/ssl/cert.pem` # MacOS / Alpine (docker container)
```
Comment (qiancai) [inline develop/dev-guide-sample-application-ruby-mysql2.md:300]:
```suggestion
1. `/etc/ssl/certs/ca-certificates.crt` for Debian, Ubuntu, Gentoo, Arch, or Slackware
2. `/etc/pki/tls/certs/ca-bundle.crt` for RedHat, Fedora, CentOS, Mageia, Vercel, or Netlify
3. `/etc/ssl/ca-bundle.pem` for OpenSUSE
4. `/etc/ssl/cert.pem` for macOS or Alpine (docker container)
```

File: develop/dev-guide-sample-application-ruby-rails.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+
+<SimpleTab>
+<div label="TiDB Serverless">
+
+1. Navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page, and then click the name of your target cluster to go to its overview page.
+
+2. Click **Connect** in the upper-right corner. A connection dialog is displayed.
+
+3. In the connection dialog, select `Rails` from the **Connect With** dropdown and keep the default setting of the **Endpoint Type** as `Public`.
+
```
[the remark this answers is not in the document]
Comment (hooopo) [reply develop/dev-guide-sample-application-ruby-rails.md:88]: 没有区别的，原因有一段写了，就是driver会自动找CA PATH

File: develop/dev-guide-sample-application-ruby-rails.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: 👌

[the remark this answers is not in the document]
Comment (qiancai) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: 那文档开头也可以加一句对 ActiveRecord ORM 的解释吗？

[the remark this answers is not in the document]
Comment (hooopo) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: 应该不需要，ActiveRecord是Rails的一个组件，大部分情况都是一起使用的：https://github.com/rails/rails/tree/main/activerecord

File: develop/dev-guide-sample-application-ruby-rails.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
+---
+title: Connect to TiDB with Rails framework and ActiveRecord ORM
+summary: Learn how to connect to TiDB using the Rails framework. This tutorial gives Ruby sample code snippets that work with TiDB using the Rails framework and ActiveRecord ORM.
+---
+
+# Connect to TiDB with Rails Framework and ActiveRecord ORM
+
+TiDB is a MySQL-compatible database, and [Rails](https://github.com/rails/rails) is a popular web application framework written in Ruby.
```
Comment (qiancai) [inline develop/dev-guide-sample-application-ruby-rails.md:8]:
```suggestion
TiDB is a MySQL-compatible database, [Rails](https://github.com/rails/rails) is a popular web application framework written in Ruby, and [ActiveRecord ORM](https://github.com/rails/rails/tree/main/activerecord) is the object-relational mapping in Rails.
```

File: develop/dev-guide-sample-application-ruby-rails.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: 因为标题里 Rails 和 ActiveRecord ORM两者都提到了，开头第一句话只介绍一个，可能会以为介绍漏了一个。你看 https://github.com/pingcap/docs/pull/14642/files#r1319409200 这样简单说下可以吗？

[the remark this answers is not in the document]
Comment (hooopo) [reply develop/dev-guide-sample-application-ruby-rails.md:2]: 👌

**Review by qiancai — APPROVED** [review]

**Pull request merged**